### PR TITLE
Trying out new branch for mobile

### PR DESCRIPTION
**Pull request by Gormiee**
new pull req no client chnagesdssxxesdwxw

### CODE REVIEW

File: src/UICrafter.Core/Components/BaseComponents/LoadMetaDataList.razor [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            try
+            {
+                // Simulate getting data from repository
+                AppViews = await AppViewRepository.GetAppViews() ?? AppViews;
+                Log.Information("Called gRPC");
+                // Simulate delay
```
Comment (Nephim) [inline src/UICrafter.Core/Components/BaseComponents/LoadMetaDataList.razor]: Fix

File: src/UICrafter.Core/Components/BaseComponents/LoadMetaDataList.razor [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    protected virtual void HandleClick(AppView view){}
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            try
+            {
+                // Simulate getting data from repository
```
Comment (Nephim) [inline src/UICrafter.Core/Components/BaseComponents/LoadMetaDataList.razor]: Fix

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 3]
```diff
@@ -34,18 +20,28 @@ else
...
-        {
-            Log.Error($"Error fetching AppViews: {ex.Message}");
-        }
-    }
+            try
+            {
+                // Simulate getting data from repository
+                AppViews = await AppViewRepository.GetAppViews() ?? AppViews;
+                Log.Information("Called gRPC");
+                // Simulate delay
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Fix

File: src/UICrafter.Core/Components/LoadMetadataComponent.razor [hunk 4]
```diff
@@ -34,18 +20,28 @@ else
...
-            // await Task.Delay(1); // Simulate delay
-        }
-        catch (Exception ex)
-        {
-            Log.Error($"Error fetching AppViews: {ex.Message}");
-        }
-    }
+            try
+            {
+                // Simulate getting data from repository
```
Comment (Nephim) [inline src/UICrafter.Core/Components/LoadMetadataComponent.razor]: Fix

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 5]
```diff
@@ -1,9 +1,17 @@
 @inject IHttpClientProvider HttpClientProvider
 
 <MudButton OnClick="CallApi"
-           Color="@buttonColor"
-           Disabled="@isDisabled">
-    @Button.Button.Label
+           Color="@buttonColor">
+    @if (_processing)
```
Comment (Nephim) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor:5]: Good job

File: src/UICrafter.Mobile/Components/Layout/NavMenu.razor [hunk 6]
```diff
@@ -8,25 +8,20 @@
...
+            <MudNavLink Href="" Match="NavLinkMatch.All" Icon="@Icons.Material.Filled.Home">Home</MudNavLink>
+            <AuthorizeView>
+                <MudNavLink Href="appview" Match="NavLinkMatch.Prefix" Icon="@Icons.Material.Filled.BugReport">AppView</MudNavLink>
+                <MudNavLink Href="authpage" Match="NavLinkMatch.Prefix" Icon="@Icons.Material.Filled.Create">Authpage</MudNavLink>
+            </AuthorizeView>
+        </MudNavMenu>
     </nav>
 </div>
+
+@* <MudNavMenu>
```
Comment (Aagaard00) [inline src/UICrafter.Mobile/Components/Layout/NavMenu.razor]: Outcommented

File: src/UICrafter.Mobile/Components/UIComponents/LoadMetaDataMobile.razor [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+@inject NavigationManager NavigationManager
+
+@{
+    base.BuildRenderTree(__builder);
+}
+
+@code {
+    protected override void HandleClick(AppView view)
+    {
+        NavigationManager.NavigateTo("/appview/" + view.Id);
```
Comment (Nephim) [inline src/UICrafter.Mobile/Components/UIComponents/LoadMetaDataMobile.razor]: Make it a Href not this

File: src/UICrafter/UICrafter.csproj [hunk 8]
```diff
@@ -42,4 +42,8 @@
     <Folder Include="Extensions\" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Include="..\UICrafter.Mobile\Components\Pages\LoadMetaDataListMobile.razor" Link="LoadMetaDataListMobile.razor" />
```
Comment (Nephim) [inline src/UICrafter/UICrafter.csproj]: This needs to be removed

**Review by Nephim — CHANGES_REQUESTED** [review]
Mostly good, there are a few things you need to fix

File: src/UICrafter.Mobile/Components/Pages/AppViewPage.razor [hunk 9]
```diff
@@ -1,43 +1,56 @@
...
+                        <PocButtonNoMistakes Button="component" />
+                        break;
+                    case UIComponent.ComponentOneofCase.InputField:
+                        <PocAPIParamInputField InputField="component" />
+                        break;
+                    case UIComponent.ComponentOneofCase.Textbox:
+                        <PocTextBox Textbox="component" />
+                        break;
+                    default:
+                        // Handle unknown components if necessary
```
Comment (Aagaard00) [inline src/UICrafter.Mobile/Components/Pages/AppViewPage.razor]: Hmmmmm

File: src/UICrafter.Mobile/Components/Pages/AppViewPage.razor [hunk 10]
```diff
@@ -48,97 +61,38 @@ else
...
-                        Content = "This is a test textbox.",
-                        SourceRef = "ButtonGUID",
-                        JsonField = "success"
-                    }
-                }
-                // Add more components as needed for testing
-            }
-                };
+        // Fetch the appView via the gRPC call
+        appView = await AppViewRepository.GetAppViewById(viewId.Value);
```
Comment (Aagaard00) [inline src/UICrafter.Mobile/Components/Pages/AppViewPage.razor:81]: Comments

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 11]
```diff
@@ -13,17 +21,17 @@
...
         var httpclient = HttpClientProvider.GetDefaultHttpClient();
         try
         {
-            isDisabled = true;
+            buttonColor = Color.Primary;
+            _processing = true;
             var originalLabel = Button.Button.Label;
-            Button.Button.Label = "Loading...";
 
             // Assume that Button.URL is the API endpoint
```
Comment (Aagaard00) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor:36]: Comment

File: src/UICrafter.Core/Components/PocButtonNoMistakes.razor [hunk 12]
```diff
@@ -34,11 +42,9 @@
...
             }
-
-            await Task.Delay(1000);
-            Button.Button.Label = originalLabel;
+            await Task.Delay(500);
             buttonColor = Color.Success;
-            isDisabled = false;
+            _processing = false;
 
             // Notify subscribers
```
Comment (Aagaard00) [inline src/UICrafter.Core/Components/PocButtonNoMistakes.razor:49]: Comment

File: src/UICrafter.Core/Utility/StupidStore.cs [hunk 13]
```diff
@@ -10,16 +10,23 @@ public void Subscribe(string buttonGUID, Action<Dictionary<string, object>> call
...
 
 	public void Unsubscribe(string buttonGUID, Action<Dictionary<string, object>> callback)
 	{
 		if (_subscribers.TryGetValue(buttonGUID, out var value))
 		{
 			value.Remove(callback);
+
+			if (value.Count == 0)
+			{
+				_subscribers.Remove(buttonGUID); // Clean up if no subscribers are left
```
Comment (Aagaard00) [inline src/UICrafter.Core/Utility/StupidStore.cs]: Comment

**Review by Aagaard00 — CHANGES_REQUESTED** [review]

**Pull request merged**